### DISCUSSION TITLE
Loop in reducer enum

### DISCUSSION THREAD

**Discussion by BakrIOS91** [pointfreeco/swift-composable-architecture#3109] [General]
if i have a

```
@Reducer
    enum FormFields {
        case fieldTextField(TieldTextFieldFeature)
        case fieldTextView(FieldTextViewFeature)
        case fieldForm(FieldFormFeature)
        case fieldUploadFile(FieldUploadFileFeature)
    }
```

what is the best way to loop on it if all will be presented inside form via CellFor( Field)

**Comment by tgrapperon** [comment]
You should probably `switch` over the enum's cases in the view (follow the example in the `NavigationStack` case study's `destination` part, it has the same shape). If your cells are in a `List`, `Form` or `ScrollView`, and if you have many of them, you should also probably wrap the whole switch in a container of some sort like `ZStack` for performance reasons (the explanation is a bit complex, but it is explained in some WWDC 2023 video about ScrollViews IIRC).

**Reply by BakrIOS91** [reply]
Thanks But do u know how to introduce in in the state and as is as state or as identified array or there a better one
